src/app/archive/lib/processor.ml

```diff
@@ -684,24 +684,35 @@ module Zkapp_permissions = struct
 
   let table_name = "zkapp_permissions"
 
-  let add_if_doesn't_exist (module Conn : CONNECTION) (perms : Permissions.t) =
+  let add_if_doesn't_exist (module Conn : CONNECTION) (perms : Permissions.t)
+      ~logger =
     let txn_version =
       Mina_numbers.Txn_version.to_int @@ snd perms.set_verification_key
     in
     let%bind versions =
       Protocol_versions.find_txn_version (module Conn) ~transaction:txn_version
     in
-    ( match versions with
-    | Ok [] ->
-        failwith
-          (sprintf "No transaction version exists for the permission, %s"
-             (Permissions.to_yojson perms |> Yojson.Safe.to_string) )
-    | Ok _ ->
-        ()
-    | Error e ->
-        failwith
-          (sprintf "fail to query protocol_versions table, see %s"
-             (Caqti_error.show e) ) ) ;
+    let%bind () =
+      match versions with
+      | Ok [] ->
+          [%log error]
+            ~metadata:[ ("perms", Permissions.to_yojson perms) ]
+            "No transaction version exists for the permission, adding a dummy \
+             version" ;
+          let%map _id =
+            Protocol_versions.add_if_doesn't_exist
```

Comment:
Presumably we should just call this in place of `Protocol_versions.find_txn_version` above? Seems odd to raise an error, because it's not actionable.

Reply:
Because we don't have a full protocol_version there. Protocol_version is a triple (txn_version, network_version, patch_version), and here we only have txn_version. That's why I am adding a dummy protocol_version with network_version and patch_version being max_int here.

Reply:
Wait, we never actually use `versions` here. Instead of adding a junk version, can we just remove all the logic around `versions` altogether?